src/app/core/session/session-service/remote-session.spec.ts

```diff
@@ -1,94 +1,52 @@
 import { environment } from "../../../../environments/environment";
+import { AuthService } from "../auth/auth.service";
 
 describe("RemoteSessionService", () => {
   let service: RemoteSession;
-  let mockHttpClient: jasmine.SpyObj<HttpClient>;
-  let dbUser: DatabaseUser;
+  let mockAuthService: jasmine.SpyObj<AuthService>;
 
   beforeEach(() => {
     environment.session_type = SessionType.mock;
-    mockHttpClient = jasmine.createSpyObj(["post", "delete"]);
-    mockHttpClient.delete.and.returnValue(of());
+    mockAuthService = jasmine.createSpyObj(["authenticate", "logout"]);
+    // Remote session allows TEST_USER and TEST_PASSWORD as valid credentials
+    mockAuthService.authenticate.and.callFake(async (u, p) => {
+      if (u === TEST_USER && p === TEST_PASSWORD) {
+        return { name: TEST_USER, roles: ["user_app"] };
+      } else {
+        throw new HttpErrorResponse({
+          status: RemoteSession.UNAUTHORIZED_STATUS_CODE,
+        });
+      }
+    });
 
     TestBed.configureTestingModule({
       providers: [
         RemoteSession,
         LoggingService,
-        { provide: HttpClient, useValue: mockHttpClient },
+        { provide: AuthService, useValue: mockAuthService },
       ],
     });
 
-    // Remote session allows TEST_USER and TEST_PASSWORD as valid credentials
-    dbUser = { name: TEST_USER, roles: ["user_app"] };
     service = TestBed.inject(RemoteSession);
-
-    mockHttpClient.post.and.callFake((url, body) => {
-      if (body.name === TEST_USER && body.password === TEST_PASSWORD) {
-        return of(dbUser as any);
-      } else {
-        return throwError(
-          new HttpErrorResponse({ status: service.UNAUTHORIZED_STATUS_CODE })
-        );
-      }
-    });
-  });
-
-  it("should be connected after successful login", async () => {
-    expect(service.loginState.value).toBe(LoginState.LOGGED_OUT);
-
-    await service.login(TEST_USER, TEST_PASSWORD);
-
-    expect(mockHttpClient.post).toHaveBeenCalled();
-    expect(service.loginState.value).toBe(LoginState.LOGGED_IN);
   });
 
   it("should be unavailable if requests fails with error other than 401", async () => {
-    mockHttpClient.post.and.returnValue(
-      throwError(new HttpErrorResponse({ status: 501 }))
+    mockAuthService.authenticate.and.rejectWith(
+      new HttpErrorResponse({ status: 501 })
     );
 
     await service.login(TEST_USER, TEST_PASSWORD);
 
     expect(service.loginState.value).toBe(LoginState.UNAVAILABLE);
   });
 
-  it("should be rejected if login is unauthorized", async () => {
-    await service.login(TEST_USER, "wrongPassword");
-
-    expect(service.loginState.value).toBe(LoginState.LOGIN_FAILED);
-  });
-
-  it("should disconnect after logout", async () => {
-    await service.login(TEST_USER, TEST_PASSWORD);
-
-    await service.logout();
-
-    expect(service.loginState.value).toBe(LoginState.LOGGED_OUT);
-  });
-
-  it("should assign the current user after successful login", async () => {
```

Comment:
I removed these tests here because they were basically testing the same things that are already covered by the abstract tests in `session-service.spec.ts`.

Reply:
ah, this is tested through the call to `testSessionServiceImplementation()` then. Great :+1: